rfcs/datepicker-component.md

```diff
@@ -0,0 +1,159 @@
+# DatePicker Component
+
+Represents control to work and pick date, time value or a range of values
+
+## Usage
+
+### Basic usage
+
+```javascript
+import * as React from 'react';
+import {DatePicker} from 'baseui/datepicker';
+
+export default () => <DatePicker />;
+```
+
+### Advanced usage
+
+```javascript
+import * as React from 'react';
+import {DatePicker} from 'baseui/datepicker';
+
+export default () => {
+  return <DatePicker
+    initialState={{
+      value: [new Date(2018, 1, 15), new Date(2018, 8, 25)]
+    }}
+    isRange={true}
```

Comment:
a range picker is essentially two individual pickers wired together as controlled components so it could be helpful to create the individual inputs first before considering range.  I just built a date range component so I can send an example if this comment is cryptic